### ISSUE TITLE
Add class imbalance data check to automl

### ISSUE THREAD

**Issue by dsherry** [alteryx/evalml#1276]
**Background**
In #971 we [implemented a data check for detecting class imbalance](https://github.com/alteryx/evalml/pull/1135/files), but we didn't add it to `DefaultDataChecks`.

**Proposal**
* Add `ClassImbalanceDataCheck` to `DefaultDataChecks`
* It currently has a warning for <10% representation. We should also have that data check include an error if the number of instances of any class is less than `2 * n_cv_folds`, because if not we'll get modeling errors. Please read the discussion in #971.
* Let's make sure our test coverage for this includes both binary and multiclass cases, categorical/int dtypes.

**Testing**
A great dataset to test with is [the abalone dataset](https://www.kaggle.com/rodolfomendes/abalone-dataset), with target "Rings", because that target has a bunch of classes which are heavily imbalanced

```python
df['Rings'].value_counts()
```
outputs
```python
9     689
10    634
8     568
11    487
7     391
12    267
6     259
13    203
14    126
5     115
15    103
16     67
17     58
4      57
18     42
19     32
20     26
3      15
21     14
23      9
22      6
24      2
27      2
1       1
25      1
2       1
26      1
29      1
Name: Rings, dtype: int64
```

**Comment by dsherry** [comment]
Note that I tried the following with [abalone.csv](https://alteryx0-my.sharepoint.com/:x:/g/personal/dylan_sherry_alteryx_com/Ee8iHh7ih6RMsrgDBy1aWvEBHr2lo5EacqkFLKN3HSjLbw?e=eJhBFF)
```python
import pandas as pd
import numpy as np
import evalml
df = pd.read_csv('abalone.csv')
y = df.pop('Rings')
automl = evalml.automl.AutoMLSearch(problem_type='multiclass')
automl.search(df, y)
```

And I got:
> /Users/dylan.sherry/.pyenv/versions/evalml/lib/python3.8/site-packages/sklearn/model_selection/_split.py:670: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=3.
> ...
> ~/development/evalml/evalml/automl/automl_search.py in _compute_cv_scores(self, pipeline, X, y)
> ...
> Exception: Missing target values in the training set after data split: {29}. Missing target values in the test set after data split: {1, 2, 26, 25}.

Because some of the CV splits contained only 1 instance of some of the classes. This is the case we're trying to avoid.
